src/backend/distributed/sql/udfs/get_missing_time_partition_ranges/latest.sql

```diff
@@ -0,0 +1,264 @@
+CREATE OR REPLACE FUNCTION pg_catalog.get_missing_time_partition_ranges(
+    table_name regclass,
+    to_date timestamptz,
+    start_from timestamptz DEFAULT now(),
+    partition_interval INTERVAL DEFAULT NULL)
+returns table(
+    partition_name text,
+    range_from_value text,
+    range_to_value text)
+LANGUAGE plpgsql
+AS $$
+DECLARE
+    distinct_partition_interval_count int;
+    table_partition_interval INTERVAL;
+    is_multiple_days boolean;
+    table_partition_column_type_name text;
+    manual_partition_from_value_text text;
+    manual_partition_to_value_text text;
+    current_range_from_value timestamptz := NULL;
+    current_range_to_value timestamptz := NULL;
+    current_range_from_value_text text;
+    current_range_to_value_text text;
+    datetime_string_format text;
+    max_table_name_length int;
+    table_name_text text;
+BEGIN
+    /* check whether the table is time partitioned table, if not error out */
+    PERFORM *
+    FROM pg_catalog.pg_partitioned_table
+    WHERE partrelid = table_name::oid;
+
+    IF NOT FOUND THEN
+        RAISE '% is not partitioned', table_name;
+    END IF;
+
+    BEGIN
+		PERFORM from_value::timestamptz, to_value::timestamptz
+		FROM pg_catalog.time_partitions
+		WHERE parent_table = table_name;
+	EXCEPTION WHEN invalid_datetime_format THEN
+		RAISE 'partition column of % cannot be cast to a timestamptz', table_name;
```

Comment:
I guess this check does not quite work if there are no partitions yet, but not sure whether there is a better way.